src/services/signer_service.ts

```diff
@@ -173,12 +214,86 @@ export class SignerService {
                     shouldValidate: false,
                 },
             );
-
+        await this._updateTransactionEntityToSubmittedAsync(txHash);
         return {
             ethereumTransactionHash,
             signedEthereumTransaction,
         };
     }
+    private async _checkForStuckTransactionsAsync(): Promise<void> {
+        const stuckTransactions = await this._transactionEntityRepository.find({
+            where: { status: TransactionStates.Stuck },
+        });
+        if (stuckTransactions.length === 0) {
+            return;
+        }
+        const gasStationPrice = await this._getGasPriceFromGasStationOrThrowAsync();
+        const targetGasPrice = gasStationPrice.multipliedBy(UNSTICKING_TRANSACTION_GAS_MULTIPLIER);
+        for (const tx of stuckTransactions) {
+            try {
+                await this._unstickTransactionAsync(tx, targetGasPrice.multipliedBy(targetGasPrice));
```

Comment:
Seems like we'd only want to unstick the transaction if its `tx.gasPrice` is < `targetGasPrice`. Otherwise we'd unstick and replace with the same  (or worse) `gasPrice`?

Reply:
Added a check, that skips the transaction where `tx.gasPrice` <= `targetGasPrice` and logs a `warning`.